clients/client/src/test/java/org/projectnessie/client/http/TestHttpClient.java

```diff
@@ -61,28 +73,88 @@ public class TestHttpClient {
 
   @InjectSoftAssertions protected SoftAssertions soft;
 
-  private static HttpRequest get(InetSocketAddress address) {
-    return get(address, false);
+  private HttpRequest get(URI baseUri) {
+    return createClient(baseUri, b -> {}).newRequest();
   }
 
-  private static HttpRequest get(InetSocketAddress address, boolean disableCompression) {
-    return get(address, 15000, 15000, disableCompression);
+  private HttpClient createClient(URI baseUri, Consumer<HttpClient.Builder> customizer) {
+    HttpClient.Builder b =
+        HttpClient.builder()
+            .setBaseUri(baseUri)
+            .setObjectMapper(MAPPER)
+            .setConnectionTimeoutMillis(15000)
+            .setReadTimeoutMillis(15000);
+    customizer.accept(b);
+    return b.build();
```

Comment:
Do we have tests for `UrlConnectionClient`?

Reply:
The build-script has a bunch of configurations, yes.